packages/beacon-node/src/api/impl/beacon/pool/index.ts

```diff
@@ -39,6 +43,9 @@ export function getBeaconPoolApi({
       // Already filtered by slot
       let attestations = chain.aggregatedAttestationPool.getAll(slot);
       const fork = chain.config.getForkName(slot ?? attestations[0]?.data.slot ?? chain.clock.currentSlot);
+      const attestationFilterPredicate = (att: Attestation): boolean =>
+        isForkPostElectra(fork) ? isElectraAttestation(att) : !isElectraAttestation(att);
+      attestations = attestations.filter(attestationFilterPredicate);
```

Comment:
I thought we were leaning more towards pruning the pool on fork transition, this will work as well but if feasible slightly lean towards pruning

Reply:
> I thought we were leaning more towards pruning the pool on fork transition, this will work as well but if feasible slightly lean towards pruning

Don't think we are doing anything special on fork transition. We just rely on `onClockSlot` to prune outdated phase0 attestations post-electra.

Reply:
I think filtering on api works just as well, just had the impression we wanna prune to avoid potential errors due to having stale (and no longer valid) attestations in the pool.

Reply:
yes even we prune in oppool, api should protect itself by implementing filter logic, so this looks good to me